tiledb/sm/filesystem/s3.cc

```diff
@@ -1465,9 +1474,16 @@ Status S3::init_client() const {
             HERE(),
             Aws::Auth::GetConfigProfileName(),
             std::chrono::minutes(60),
-            [client_config](const auto& credentials) {
+            [config = this->client_config_](const auto& credentials) {
               return make_shared<Aws::STS::STSClient>(
-                  HERE(), credentials, client_config);
+                  HERE(),
+                  credentials,
+                  // Create default endpoint provider.
+                  make_shared<Aws::STS::STSEndpointProvider>(HERE()),
+                  // *config is an S3ClientConfiguration&, but gets
+                  // casted to ClientConfiguration and copied to the STS
+                  // client configuration.
+                  Aws::STS::STSClientConfiguration(*config));
```

Comment:
`STSClientConfiguration` does not specify any additional parameters over `ClientConfiguration`.

We have to make a copy but I don't think it matters since this lambda is being called when the token is being refreshed, which is measured in minutes.